action.js

```diff
@@ -23,27 +23,33 @@ import { memoize } from "./memo.js";
  *      single, object argument into a list of sorted values
  */
 const makeAction = (type, payloadKeys) => {
-  payloadKeys.sort();
-  const memoized = memoize((...values) => {
-    const payload = {};
-    values.forEach((v, i) => {
-      payload[payloadKeys[i]] = v;
+  if (payloadKeys) {
+    payloadKeys.sort();
+    const memoized = memoize((...values) => {
+      const payload = {};
+      values.forEach((v, i) => {
+        payload[payloadKeys[i]] = v;
+      });
+      return { type, payload };
     });
-    return { type, payload };
-  });
-  return (obj) => memoized(...payloadKeys.map((k) => obj[k]));
+    return (obj) => memoized(...payloadKeys.map((k) => obj[k]));
+  } else {
+    return (ary) => {
+      return { type, payload: ary };
```

Comment:
@jreut This might be a little hacky. I avoided memoizing here because it was a deep rabbit hole that seemed endless for me. Seems like we might _want_ to memoize each object inside the outermost array but I also don't honestly know why it's vitally important to do. What do you think?